src/hooks/balances/resolvers/query/balances.test.tsx

```diff
@@ -0,0 +1,272 @@
+import {
+  balancesByAddressQueryResolverFactory,
+  BalancesByAddressResolverArgs,
+  Entity,
+  objectToArrayWithoutNull,
+  useBalanceQueryResolvers,
+} from './balances';
+
+import { AssetIds, Balance } from '../../../../generated/graphql';
+import { Resolvers, useQuery } from '@apollo/client';
+import { MockedProvider } from '@apollo/client/testing';
+import TestRenderer, { act } from 'react-test-renderer';
+import { gql } from 'graphql.macro';
+import {
+  getMockApiPromise,
+  nativeAssetBalance,
+  nonNativeAssetBalance,
+} from '../../../polkadotJs/tests/mockUsePolkadotJsContext';
+import { ApiPromise } from '@polkadot/api';
+import waitForExpect from 'wait-for-expect';
+import errors from '../../../../errors';
+waitForExpect.defaults.interval = 1;
+
+jest.mock('../../../polkadotJs/usePolkadotJs', () => ({
+  usePolkadotJsContext: () => {
+    // jest executes before imports, that's why require is here
+    const {
+      mockUsePolkadotJsContext,
+    } = require('../../../polkadotJs/tests/mockUsePolkadotJsContext');
+    return mockUsePolkadotJsContext();
+  },
+}));
+
+describe('hooks/balances/resolvers/query/balances', () => {
+  describe('objectToArrayWithoutNull', () => {
+    it('can convert an object to an array', () => {
+      const assetIds = {
+        a: '0',
+        b: '1',
+      };
+      const assets = objectToArrayWithoutNull(assetIds);
```

Comment:
you could effectively filter out `undefined` as well, if in your implementation you'd just do:
```
.filter(value => value)
```

Reply:
FYI that is not necessary, the implementation also filters out undefined. There is also a test for it called 'can convert an object to an array with one undefined value'.
I will change the name of the function to reflect this behaviour.